### ISSUE TITLE
File Locking paths are not cross platforms

### ISSUE THREAD

**Issue by SRombauts** [git-lfs/git-lfs#2138]
Trying to lock the same file from Windows 10 command line and Ubuntu "Bash" on Windows 10 version 1703 (Creators Update) from within the same clone results in two successful different/redundant locks!

Using a https://github.com remote public project for testing purpose.

1. From Ubuntu "Bash" command line:

```
srombauts@ASUS-SEB:/mnt/c/Unreal Projects/MyProject$ git lfs lock Content/MyProject/Actor_BP.uasset
Locked Content/MyProject/Actor_BP.uasset

srombauts@ASUS-SEB:/mnt/c/Unreal Projects/MyProject$ git lfs locks
Content/MyProject/Actor_BP.uasset       SRombauts       ID:1150
```

2. From Git for Windows MINGW64 (or standard Windows) command line:

```
sebas@ASUS-SEB MINGW64 /c/Unreal Projects/MyProject (master)
$ git lfs lock Content/MyProject/Actor_BP.uasset
Locked Content/MyProject/Actor_BP.uasset

sebas@ASUS-SEB MINGW64 /c/Unreal Projects/MyProject (master)
$ git lfs locks
Content/MyProject/Actor_BP.uasset       SRombauts       ID:1150
Content\MyProject\Actor_BP.uasset       SRombauts       ID:1151
```

3. And results are symmetricals: 

```
srombauts@ASUS-SEB:/mnt/c/Unreal Projects/MyProject$ git lfs locks
Content/MyProject/Actor_BP.uasset       SRombauts       ID:1150
Content\MyProject\Actor_BP.uasset       SRombauts       ID:1151
```

I've done multiple tests with variants (orders of locks, using different command line windows...) but everything looks like locking path are client dependents.

Best Regards,
Sébastien

**Comment by ttaylorr** [comment]
@SRombauts thanks for opening this in such detail, and sorry that you ran into this. This is a bug in the LFS client where filepaths were not normalized based on OS-specific path separators.

I went ahead and fixed this in #2139 which should ship in LFS v2.1.0.

**Comment by SRombauts** [comment]
Thanks !
